controllers/datadogagent/override/global.go

```diff
@@ -6,16 +6,201 @@
+			}
+		}
+	}
+
+	if componentName == v2alpha1.NodeAgentComponentName {
+		// Tags contains a list of tags to attach to every metric, event and service check collected.
+		if config.Tags != nil {
+			tags, _ := json.Marshal(config.Tags)
+			manager.EnvVar().AddEnvVar(&corev1.EnvVar{
+				Name:  apicommon.DDTags,
+				Value: string(tags),
+			})
+		}
+
+		// Provide a mapping of Kubernetes Labels to Datadog Tags.
+		if config.PodLabelsAsTags != nil {
+			podLabelsAsTags, _ := json.Marshal(config.PodLabelsAsTags)
+			manager.EnvVar().AddEnvVar(&corev1.EnvVar{
+				Name:  apicommon.DDPodLabelsAsTags,
+				Value: string(podLabelsAsTags),
+			})
+		}
+
+		// Provide a mapping of Kubernetes Annotations to Datadog Tags.
+		if config.PodAnnotationsAsTags != nil {
+			podAnnotationsAsTags, _ := json.Marshal(config.PodAnnotationsAsTags)
+			manager.EnvVar().AddEnvVar(&corev1.EnvVar{
+				Name:  apicommon.DDPodAnnotationsAsTags,
+				Value: string(podAnnotationsAsTags),
+			})
+		}
+
+		// LocalService contains configuration to customize the internal traffic policy service.
+		gitVersion := resourcesManager.Store().GetVersionInfo()
+		if utils.IsAboveMinVersion(gitVersion, minLocalServiceVersion) {
+			if config.LocalService != nil {
+				if apiutils.BoolValue(config.LocalService.ForceEnableLocalService) || utils.IsAboveMinVersion(gitVersion, minDefaultLocalServiceVersion) {
+					if config.LocalService.NameOverride != nil {
+						_ = resourcesManager.ServiceManager().BuildAgentLocalService(dda, *config.LocalService.NameOverride)
+					} else {
+						_ = resourcesManager.ServiceManager().BuildAgentLocalService(dda, "")
+					}
+				}
+			}
+		}
+
+		// Kubelet contains the kubelet configuration parameters.
+		if config.Kubelet != nil {
+			var kubeletHostValueFrom *corev1.EnvVarSource
+			if config.Kubelet.Host != nil {
+				kubeletHostValueFrom = config.Kubelet.Host
+			} else {
+				kubeletHostValueFrom = &corev1.EnvVarSource{
+					FieldRef: &corev1.ObjectFieldSelector{
+						FieldPath: apicommon.FieldPathStatusHostIP,
+					},
+				}
+			}
+			manager.EnvVar().AddEnvVar(&corev1.EnvVar{
+				Name:      apicommon.DDKubeletHost,
+				ValueFrom: kubeletHostValueFrom,
+			})
+
+			if config.Kubelet.TLSVerify != nil {
+				manager.EnvVar().AddEnvVar(&corev1.EnvVar{
+					Name:  apicommon.DDKubeletTLSVerify,
+					Value: apiutils.BoolToString(config.Kubelet.TLSVerify),
+				})
+			}
+
+			manager.EnvVar().AddEnvVar(&corev1.EnvVar{
+				Name:  apicommon.DDKubeletCAPath,
+				Value: config.Kubelet.AgentCAPath,
+			})
+
+			if config.Kubelet.HostCAPath != "" {
+				kubeletVol, kubeletVolMount := volume.GetVolumes(apicommon.KubeletCAVolumeName, config.Kubelet.HostCAPath, config.Kubelet.AgentCAPath, true)
+				manager.VolumeMount().AddVolumeMountToContainers(
+					&kubeletVolMount,
+					[]apicommonv1.AgentContainerName{
+						apicommonv1.CoreAgentContainerName,
+						apicommonv1.ProcessAgentContainerName,
+						apicommonv1.TraceAgentContainerName,
```

Comment:
just noting that in case one of these containers doesn't exist, nothing will happen